test/controllers/analytics.js

```diff
@@ -0,0 +1,65 @@
+const request = require('supertest');
```

Comment:
This is out of scope for this PR so don't worry about it for now, but it would be nice to rewrite the tests so that we don't need to use the `done` callback.

Reply:
can you provide me a tutorial/information link? please ;)

Reply:
I don't know a good resource to learn, but you can try this. Make sure you `await` the request, otherwise your test will give you a false positive (it will pass when it shouldn't).

```javascript
it('does something', async function() {
  const res = await request(server)
    .post('/my/route')
    .set('Authorization', `Bearer ${authToken}`)
    .set('Accept', 'application/json')
    .send({})
    .expect('Content-Type', /json/)
    .expect(200);
  expect(res.body.someProperty).to.equal('something');
});
```

(But let's see if @martinbedouret likes this approach first.)

Reply:
@tomivm @sylvansson I like this approach as we work like synchronous test and we forget about the `done`  callbacks.. Good suggestion!!

Reply:
thanks you very much for your explanations! I will study more and start a new PR

Reply:
thanks very much for yours explanations! I gonna try/study this and start a new PR